### PR TITLE
feat: Upgrade Mobile App to Latest

### PR DESCRIPTION
**Pull request by rupam-seal**
### <samp>🤖 Generated by Copilot at ef04787</samp>

This pull request updates the `package.json` file to use the latest versions of Ionic, Angular, Capacitor, and other dependencies and dev dependencies. This ensures that the `fyle-mobile-app` project can benefit from the latest improvements and security fixes of the frameworks and libraries it relies on.

### <samp>🤖 Generated by Copilot at ef04787</samp>

> _Sing, O Muse, of the valiant Fyle-mobile-app,_
> _That with Ionic, Angular, and Capacitor wrought_
> _A splendid interface for managing expenses,_
> _And native features for each platform it supports._

### <samp>🤖 Generated by Copilot at ef04787</samp>

*  Update versions of Angular, Ionic, Capacitor, Sentry, and other core dependencies for the project ([link](https://github.com/fylein/fyle-mobile-app/pull/2237/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L25-R70))
*  Update versions of dayjs, dotenv, heic2any, ng2-pdf-viewer, ngx-image-cropper, ngx-pinch-zoom-16, rxjs, swiper, tslib, and zone.js, dependencies that provide specific features and functionality for the app ([link](https://github.com/fylein/fyle-mobile-app/pull/2237/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L77-R111))
*  Update version of husky, a dev dependency that enables git hooks for linting, formatting, and testing the code ([link](https://github.com/fylein/fyle-mobile-app/pull/2237/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L117-R117))
*  Update versions of prettier, ts-node, and typescript, dev dependencies that are used for code formatting, TypeScript execution, and compilation ([link](https://github.com/fylein/fyle-mobile-app/pull/2237/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L127-R130))

## Clickup
https://app.clickup.com/t/85ztp0vrq

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ const routes: Routes = [
 export const appRoutes = routes;
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes, { preloadingStrategy: PreloadAllModules, relativeLinkResolution: 'legacy' })],
```
Comment (rupam-seal) [inline src/app/app-routing.module.ts:36]: relativeLinkResolution is in app-routing.module.ts
@NgModule({
  imports: [RouterModule.forRoot(routes, { preloadingStrategy: PreloadAllModules, relativeLinkResolution: 'legacy' })],
  exports: [RouterModule],
})
 relativeLinkResolution is deprecated. In v16, relativeLinkResolution is no longer configurable in the Router.
[ExtraOptions](https://angular.io/api/router/ExtraOptions)
[PR #47623](https://github.com/angular/angular/pull/47623)

we can remove the relativeLinkResolution.
sof: [link](https://stackoverflow.com/a/76171393)

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ const routes: Routes = [
...
+  imports: [RouterModule.forRoot(routes, { preloadingStrategy: PreloadAllModules })],
```
Comment (rupam-seal) [inline src/app/app-routing.module.ts:36]: relativeLinkResolution is in app-routing.module.ts
```
@NgModule({
  imports: [RouterModule.forRoot(routes, { preloadingStrategy: PreloadAllModules, relativeLinkResolution: 'legacy' })],
  exports: [RouterModule],
})
```
 relativeLinkResolution is deprecated. In v16, relativeLinkResolution is no longer configurable in the Router.
[ExtraOptions](https://angular.io/api/router/ExtraOptions)
[PR #47623](https://github.com/angular/angular/pull/47623)

we can remove the relativeLinkResolution.
sof: [link](https://stackoverflow.com/a/76171393)

File: src/app/app.module.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ import { GooglePlus } from '@awesome-cordova-plugins/google-plus/ngx';
 import { InAppBrowser } from '@awesome-cordova-plugins/in-app-browser/ngx';
 import { SharedModule } from './shared/shared.module';
 import { CurrencyPipe, TitleCasePipe } from '@angular/common';
-import * as Sentry from '@sentry/angular';
+import * as Sentry from '@sentry/angular-ivy';
```
Comment (rupam-seal) [inline src/app/app.module.ts:14]: Screenshot 2023-07-26 at 12 12 49 PM

The error indicates that @Sentry/angular@7.60.0 is not compatible with Angular 16. To resolve this, we will need to find a version of @Sentry/angular that has a peer dependency on @angular/common version ^16.0.0.
The [@Sentry/angular-ivy](https://www.npmjs.com/package/@sentry/angular-ivy) SDK officially supports Angular 12 to 16.

File: src/app/core/services/device.service.ts [hunk 4]
```diff
@@ -21,10 +21,11 @@ export class DeviceService {
     }).pipe(
       map(({ deviceInfo, deviceId, appInfo }) =>
         Object.assign(deviceInfo, deviceId, {
+          uuid: '',
```
Comment (rupam-seal) [inline src/app/core/services/device.service.ts:24]: Screenshot 2023-08-08 at 12 01 50 PM

File: src/app/fyle/my-view-report/my-view-report.page.ts [hunk 5]
```diff
@@ -514,7 +515,7 @@ export class MyViewReportPage {
 
   segmentChanged(event: SegmentCustomEvent): void {
     if (event?.detail?.value) {
-      this.segmentValue = parseInt(event.detail.value, 10);
+      this.segmentValue = parseInt(String(event.detail.value), 10);
```
Comment (rupam-seal) [inline src/app/fyle/my-view-report/my-view-report.page.ts:518]: Screenshot 2023-08-08 at 12 08 07 PM

File: src/app/fyle/split-expense/split-expense.page.ts [hunk 6]
```diff
@@ -626,7 +626,7 @@ export class SplitExpensePage implements OnInit {
       txnDt = dateOfTxn ? new Date(dateOfTxn) : today;
       txnDt = dayjs(txnDt).format('YYYY-MM-DD');
     }
-    const fg = this.formBuilder.group({
+    const fg: FormGroup<any> = this.formBuilder.group({
```
Comment (rupam-seal) [inline src/app/fyle/split-expense/split-expense.page.ts:629]: Screenshot 2023-08-08 at 12 11 48 PM

File: src/app/shared/components/dependent-fields/dependent-fields.component.ts [hunk 7]
```diff
@@ -155,9 +158,9 @@ export class DependentFieldsComponent implements OnInit, OnDestroy, OnChanges {
     this.dependentFields = this.dependentFields.slice(0, updatedFieldIndex + 1);
   }
 
-  onDependentFieldChanged(data: { id: number; label: string; parent_field_id: number; value: string }): void {
+  onDependentFieldChanged(data: Partial<{ id: number; label: string; parent_field_id: number; value: string }>): void {
```
Comment (rupam-seal) [inline src/app/shared/components/dependent-fields/dependent-fields.component.ts:161]: Screenshot 2023-08-08 at 12 14 41 PM

File: src/theme/variables.scss [hunk 8]
```diff
@@ -10,7 +10,7 @@
 $custom-typography: mat.define-typography-config(
   $font-family: '"aktiv-grotesk", "Lato", sans-serif',
 );
-@include mat.core($custom-typography);
```
Comment (rupam-seal) [inline src/theme/variables.scss:13]: Screenshot 2023-08-08 at 12 18 01 PM

- With version 16, we can no longer pass a typography parameter when calling core() in variables.scss with $custom-typography.

- So, I removed the argument from @include mat.core(); and updated the $fy-theme

from:
`$fy-theme: mat.define-light-theme($fy-theme-primary, $fy-theme-accent, $fy-theme-warn, $custom-typography);`
to:
```
$fy-theme: mat.define-light-theme((
  color: (
    primary: $fy-theme-primary,
    accent: $fy-theme-accent,
    warn: $fy-theme-warn
  ),
  typography: $custom-typography,
 ));
```
as per the docs ([material typography](https://material.angular.io/guide/typography))

File: src/test.ts [hunk 9]
```diff
@@ -10,7 +10,3 @@ declare const require: any;
 getTestBed().initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting(), {
   teardown: { destroyAfterEach: false },
 });
-// Then we find all the tests.
-const context = require.context('./', true, /\.spec\.ts$/);
-// And load the modules.
-context.keys().map(context);
```
Comment (rupam-seal) [inline src/test.ts:16]: Screenshot 2023-08-08 at 1 25 02 PM

sof: https://stackoverflow.com/questions/74546242/karma-jasmine-uncaught-typeerror-webpack-require-context-is-not-a-fu/74572172#74572172

File: browserslist [hunk 10]
```diff
@@ -1,12 +1,6 @@
...
-last 2 versions
-Firefox ESR
-not dead
-not IE 9-11 # For IE 9-11 support, remove 'not'.
+Chrome >=79
+ChromeAndroid >=79
+Firefox >=70
+Edge >=79
+Safari >=14
+iOS >=14
```
Comment (rupam-seal) [inline browserslist:6]: https://ionicframework.com/docs/updating/7-0#browser-support

File: ios/.gitignore [hunk 11]
```diff
@@ -5,7 +5,7 @@
 App/build
 App/Pods
 App/App/public
-App/Podfile.lock
+App/output
```
Comment (rupam-seal) [inline ios/.gitignore:8]: https://capacitorjs.com/docs/updating/5-0#update-gitignore

File: android/variables.gradle [hunk 12]
```diff
@@ -1,21 +1,21 @@
...
+    androidxEspressoCoreVersion = '3.5.1'
     cordovaAndroidVersion = '10.1.1'
-    androidxActivityVersion = '1.4.0'
+    androidxActivityVersion = '1.7.0'
     androidxCoordinatorLayoutVersion = '1.2.0'
-    androidxFragmentVersion = '1.4.1'
+    androidxFragmentVersion = '1.5.6'
     coreSplashScreenVersion = '1.0.0-rc01'
-    androidxWebkitVersion = '1.4.0'
+    androidxWebkitVersion = '1.6.1'
```
Comment (rupam-seal) [inline android/variables.gradle:20]: https://capacitorjs.com/docs/updating/5-0#update-android-project-variables
https://capacitorjs.com/docs/updating/5-0#browser
https://capacitorjs.com/docs/updating/5-0#geolocation

File: android/build.gradle [hunk 13]
```diff
@@ -7,8 +7,8 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.2.1'
-        classpath 'com.google.gms:google-services:4.3.13'
```
Comment (rupam-seal) [inline android/build.gradle:11]: https://capacitorjs.com/docs/updating/5-0#update-google-services

File: android/build.gradle [hunk 14]
```diff
@@ -7,8 +7,8 @@ buildscript {
...
+        classpath 'com.android.tools.build:gradle:8.0.0'
+        classpath 'com.google.gms:google-services:4.3.15'
```
Comment (rupam-seal) [inline android/build.gradle:11]: https://capacitorjs.com/docs/updating/5-0#update-google-services

File: android/build.gradle [hunk 15]
```diff
@@ -7,8 +7,8 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.2.1'
-        classpath 'com.google.gms:google-services:4.3.13'
+        classpath 'com.android.tools.build:gradle:8.0.0'
```
Comment (rupam-seal) [inline android/build.gradle:10]: https://capacitorjs.com/docs/updating/5-0#update-gradle-plugin-to-800

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 16]
```diff
@@ -1,6 +1,6 @@
 #Sat Apr 23 20:33:12 IST 2022
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0.2-all.zip
```
Comment (rupam-seal) [inline android/gradle/wrapper/gradle-wrapper.properties:3]: https://capacitorjs.com/docs/updating/5-0#update-gradle-wrapper-to-802

File: android/gradle.properties [hunk 17]
```diff
@@ -19,6 +19,4 @@ org.gradle.jvmargs=-Xmx1536m
 # AndroidX package structure to make it clearer which packages are bundled with the
 # Android operating system, and which are packaged with your app's APK
 # https://developer.android.com/topic/libraries/support-library/androidx-rn
-android.useAndroidX=true
-# Automatically convert third-party libraries to use AndroidX
-android.enableJetifier=true
```
Comment (rupam-seal) [inline android/gradle.properties:24]: https://capacitorjs.com/docs/updating/5-0#update-gradle-wrapper-to-802

File: android/app/src/main/AndroidManifest.xml [hunk 18]
```diff
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
-    package="com.ionicframework.fyle595781">
```
Comment (rupam-seal) [inline android/app/src/main/AndroidManifest.xml:4]: https://capacitorjs.com/docs/updating/5-0#move-package-to-buildgradle

File: android/app/src/main/AndroidManifest.xml [hunk 19]
```diff
@@ -1,7 +1,6 @@
...
+    xmlns:tools="http://schemas.android.com/tools">
```
Comment (rupam-seal) [inline android/app/src/main/AndroidManifest.xml:3]: https://capacitorjs.com/docs/updating/5-0#move-package-to-buildgradle

File: android/app/build.gradle [hunk 20]
```diff
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.application'
 
 android {
+    namespace "com.ionicframework.fyle595781"
```
Comment (rupam-seal) [inline android/app/build.gradle:4]: https://capacitorjs.com/docs/updating/5-0#move-package-to-buildgradle

File: android/app/src/main/AndroidManifest.xml [hunk 21]
```diff
@@ -81,4 +80,5 @@
     <uses-permission android:name="android.permission.WAKE_LOCK" tools:node="remove" />
     <uses-permission android:name="android.permission.VIBRATE" tools:node="remove" />
     <uses-permission android:name="android.permission.GET_ACCOUNTS" tools:node="remove" />
+    <uses-permission android:name="android.permission.READ_MEDIA_IMAGES"/>
```
Comment (rupam-seal) [inline android/app/src/main/AndroidManifest.xml:83]: https://capacitorjs.com/docs/updating/5-0#camera

File: src/app/core/services/device.service.ts (hunk 4 above)
Comment (Julias0) [reply src/app/core/services/device.service.ts:24]: Please fix this based on our conversation yesterday

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comments

**Review by Dimple16 — COMMENTED** [review]
this is already being reviewed, will skip

**Review by Julias0 — APPROVED** [review]

**Review by OmkarJ13 — APPROVED** [review]
Please check and test for any breaking changes

**Pull request closed without merging**